analysis/filter/StructureModifier/src/main/kotlin/de/maibornwolff/codecharta/filter/structuremodifier/ParserDialog.kt

```diff
@@ -15,40 +16,58 @@ class ParserDialog {
+                            StructureModifierAction.MOVE_NODES.descripton,
+                            StructureModifierAction.REMOVE_NODES.descripton
+                    )
             )
 
-            val setRoot: String =
-                    KInquirer.promptInput(message = "What path within project to be extracted? (Optional)", default = "")
+            return when (selectedAction) {
+                StructureModifierAction.PRINT_STRUCTURE.descripton -> listOf(inputFolderName, *collectPrintArguments())
+                StructureModifierAction.SET_ROOT.descripton -> listOf(inputFolderName, *collectSetRootArguments())
+                StructureModifierAction.MOVE_NODES.descripton -> listOf(inputFolderName, *collectMoveNodesArguments())
+                StructureModifierAction.REMOVE_NODES.descripton -> listOf(inputFolderName, *collectRemoveNodesArguments())
+                else -> listOf()
+            }
+        }
 
+        private fun collectPrintArguments(): Array<String> {
             val printLevels: BigDecimal =
                     KInquirer.promptInputNumber(message = "How many print levels do you want to print? (Optional)", default = "0", hint = "0")
+            return arrayOf("--print-levels=$printLevels")
+        }
 
-            val moveFrom: String = KInquirer.promptInput(
-                    message = "What are the nodes to be moved? (Optional)"
-            )
-            var moveTo: String = ""
-            if (moveFrom.isNotBlank()) {
-                moveTo = KInquirer.promptInput(
-                        message = "Where to move them?"
-                )
-            }
-            val remove: String = KInquirer.promptInput(
-                    message = "What are the nodes to be removed? (Optional)"
-            )
-            return listOfNotNull(
-                    inputFolderName,
-                    "--output-file=$outputFileName",
-                    if (printLevels.toInt() != 0) "--print-levels=$printLevels" else null,
-                    if (setRoot.isNotBlank()) "--set-root=$setRoot" else null,
-                    if (moveFrom.isNotBlank()) "--move-from=$moveFrom" else null,
-                    if (moveFrom.isNotBlank()) "--move-to=$moveTo" else null,
-                    if (remove.isNotBlank()) "--remove=$remove" else null,
-            )
+        private fun collectSetRootArguments(): Array<String> {
+            val setRoot: String =
+                    KInquirer.promptInput(message = "What path within project to be extracted?")
+            val outputFileName = collectOutputFileName()
+            return arrayOf("--set-root=$setRoot", "--output-file=$outputFileName")
+        }
+
+        private fun collectMoveNodesArguments(): Array<String> {
+            val moveFrom: String =
+                    KInquirer.promptInput(message = "What are the nodes to be moved?")
+            val moveTo: String =
+                    KInquirer.promptInput(message = "Where to move them?")
```

Comment:
```suggestion
                    KInquirer.promptInput(message = "What is the target path to move them?")
```

Should we explain what happens if the target path does (not) exist?

Reply:
I added an explanation to the documentation